### PR TITLE
Maintain and upload artifacts index

### PR DESCRIPTION
**Pull request by agners**
Make the artifacts browsable by maintaining a list of builds. This keeps it up-to-date even when deleting images from the object storage, and minimizes queries to the object storage.

### CODE REVIEW

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -202,15 +202,35 @@ jobs:
...
+        run: |
+          aws s3api list-objects-v2
+            --bucket "${{ secrets.R2_OS_ARTIFACTS_BUCKET }}" \
+            --endpoint-url ${{ secrets.R2_OS_ARTIFACTS_ENDPOINT }}
+            --query 'Contents[].Key' | jq 'map(split("/")[0]) | unique | sort' > index.json
+          aws s3 sync \
+            ./ \
+            s3://${{ secrets.R2_OS_ARTIFACTS_BUCKET }}/ \
+            --exclude "*" \
+            --include "index.json" \
```
[the remark this answers is not in the document]
Comment (agners) [reply .github/workflows/build.yaml:233]: Hm, kinda annoying :cry:  Can't this be done with some cache control attribute? E.g. only force refresh if a browser forces refresh through cache control headers? :thinking:

[the remark this answers is not in the document]
Comment (ludeeus) [reply .github/workflows/build.yaml]: Not reliably, but its a single file, jst bust it as a last step.

[the remark this answers is not in the document]
Comment (agners) [reply .github/workflows/build.yaml]: I've added the cache flush. This requires secrets update still.

File: .github/os-artifacts/index.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                        const a = document.createElement('a');
+                        a.href = url;
+                        a.textContent = image;
+                        li.appendChild(a);
+                        ul.appendChild(li);
+                    });
+
+                    osBuildsList.appendChild(ul);
+                } catch (error) {
+                    console.error('Error fetching images:', error);
```
[the remark this answers is not in the document]
Comment (agners) [reply .github/os-artifacts/index.html:75]: Added error handling when response is not ok.

**Review by ludeeus — APPROVED** [review]

**Pull request merged**